queue_services/payment-reconciliations/src/reconciliations/eft/eft_reconciliation.py

```diff
@@ -416,48 +420,37 @@ def _shortname_balance_as_dict(eft_transactions: List[EFTRecord]) -> Dict:
 
         shortname = eft_transaction.transaction_description
         transaction_date = eft_transaction.transaction_date
+        deposit_amount = eft_transaction.deposit_amount_cad / 100
+        transaction = {'id': eft_transaction.id, 'deposit_amount': deposit_amount}
 
         if shortname in shortname_balance:
             shortname_balance[shortname]['transaction_date'] = transaction_date
-            shortname_balance[shortname]['balance'] += eft_transaction.deposit_amount_cad / 100
+            shortname_balance[shortname]['balance'] += deposit_amount
+            shortname_balance[shortname]['transactions'].append(transaction)
         else:
             shortname_balance[shortname] = {'transaction_date': transaction_date,
-                                            'balance': eft_transaction.deposit_amount_cad / 100}
-
+                                            'balance': deposit_amount,
+                                            'transactions': [transaction]}
```

Comment:
note for future:

`.setdefault(key, []).append(val)`

Reply:
Updated. Thanks!